dags/main_summary_backfill_mar_2018.py

```diff
@@ -0,0 +1,170 @@
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.models import DagBag
+from airflow.utils.trigger_rule import TriggerRule
+from airflow.contrib.operators.emr_terminate_job_flow_operator import EmrTerminateJobFlowOperator
+from airflow.contrib.sensors.emr_step_sensor import EmrStepSensor
+from airflow.operators.sensors import TimeDeltaSensor
+from airflow.operators.subdag_operator import SubDagOperator
+from airflow.operators.moz_emr import (EmrAddStepsOperator,
+                                       EmrCreateJobFlowSelectiveTemplateOperator,
+                                       MozEmrClusterStartSensor,
+                                       MozEmrClusterEndSensor)
+
+from utils.tbv import tbv_envvar
+
+
+SCHEDULE_INTERVAL = '@weekly'
+START_DATE = datetime(2017, 3, 17)
+
+default_args = {
+    'owner': 'ssuh@mozilla.com',
+    'depends_on_past': False,
+    'start_date': START_DATE,
+    'email': ['telemetry-alerts@mozilla.com', 'ssuh@mozilla.com', 'frank@mozilla.com'],
+    'email_on_failure': True,
+    'email_on_retry': True,
+    'retries': 2,
+    'retry_delay': timedelta(minutes=15)
+}
+
+
+dag = DAG('main_summary_backfill_mar_2018',
+          default_args=default_args,
+          schedule_interval=SCHEDULE_INTERVAL,
+          catchup=True,
+          # Change this back to 10 after we get a successful run on prod
+          max_active_runs=1)
+
+job_flow_id_template = "{{ task_instance.xcom_pull('setup_backfill_cluster', key='return_value') }}"
+
+
+# workaround to retry the entire subdag on step failure instead of just that step
+# via https://stackoverflow.com/a/49029261/537248
+def clear_subdag_callback(context):
+    dag_id = "{}.{}".format(context['dag'].dag_id, context['ti'].task_id)
+    execution_date = context['execution_date']
+    sdag = DagBag().get_dag(dag_id)
+    sdag.clear(
+        start_date=execution_date,
+        end_date=execution_date,
+        only_failed=False,
+        only_running=False,
+        confirm_prompt=False,
+        include_subdags=False)
+
+
+def main_summary_subdag_factory(parent_dag, task_id, day):
+    ds = "{{{{ macros.ds_format(macros.ds_add(ds, {0}), '%Y-%m-%d', '%Y%m%d') }}}}".format(day)
```

Comment:
It's because that's how you escape curly braces for `format` (I'll note this ate a good half hour or more for me trying to figure out wtf was happening)

Reply:
That's unpleasant.